modules/transactions/src/test/java/org/apache/ignite/internal/tx/storage/state/test/TestConcurrentHashMapTxStateStorage.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.tx.storage.state.test;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.apache.ignite.lang.ErrorGroups.Transactions.TX_STATE_STORAGE_DESTROY_ERR;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import org.apache.ignite.internal.tx.TxMeta;
+import org.apache.ignite.internal.tx.TxState;
+import org.apache.ignite.internal.tx.storage.state.TxStateStorage;
+import org.apache.ignite.internal.util.Cursor;
+import org.apache.ignite.lang.IgniteBiTuple;
+import org.apache.ignite.lang.IgniteInternalException;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Test implementation of {@link TxStateStorage} based on {@link ConcurrentHashMap}.
+ */
+public class TestConcurrentHashMapTxStateStorage implements TxStateStorage {
+    private final ConcurrentHashMap<UUID, TxMeta> storage = new ConcurrentHashMap<>();
+
+    private volatile boolean isStarted;
+
+    /** Snapshot restore lock. */
+    private final Object snapshotRestoreLock = new Object();
+
+    private final Map<Path, List<Map.Entry<UUID, TxMeta>>> snapshots = new ConcurrentHashMap<>();
```

Comment:
Well, snapshot logic isn't implemented currently even within persistent storages, however we either should throw UnsupportedOperationException for onSnapshotSave/Load or support transferable snapshots. Meaning that snapshot might be used in order to perform raft rebalance. If u'll stick with second option please check ConcurrentMapClusterStateStorage appoach based on SNAPSHOT_FILE.